be/src/storage/tablet_reader.cpp

```diff
@@ -283,6 +283,7 @@ Status TabletReader::get_segment_iterators(const TabletReaderParams& params, std
     rs_opts.rowid_range_option = params.rowid_range_option;
     rs_opts.short_key_ranges_option = params.short_key_ranges_option;
     rs_opts.asc_hint = _is_asc_hint;
+    rs_opts.prune_column_after_index_filter = params.prune_column_after_index_filter;
```

Comment:
Does lake OlapTable also support this? If so, `storage/lake/tablet_reader.cpp` and `storage/lake/rowset.cpp` also need pass this.

Reply:
Currently, GIN only used for OlapTable but not lake. And this optimization can only used for GIN. So lake table will not affect by this for now